src/parallel/ocbbs.cpp

```diff
@@ -1171,157 +1191,153 @@ char* BBSImpl::execute_helper(size_t* size, int id, bool exec) {
                     break;
                 }
                 ob = nullptr;
             }
             if (!ob) {
-                fprintf(stderr, "%s[%d] is not an Object in this process\n", s, i);
+                fprintf(stderr,
+                        "%s[%d] is not an Object in this process\n",
+                        template_name,
+                        object_index);
                 hoc_execerror("ParallelContext execution error", 0);
             }
-            delete[] s;
-            s = upkstr();
-            fname = hoc_table_lookup(s, sym->u.ctemplate->symtable);
+            delete[] template_name;
+            char* fname_str = upkstr();
+            fname = hoc_table_lookup(fname_str, sym->u.ctemplate->symtable);
+            if (!fname) {
+                fprintf(stderr, "%s not a function in %s\n", fname_str, hoc_object_name(ob));
+                hoc_execerror("ParallelContext execution error", 0);
+            }
+            delete[] fname_str;
             if (subworld) {
                 hoc_execerror("with subworlds, this submit style not implemented", 0);
             }
         } else if (style == 3) {  // Python callable
-            s = upkpickle(&npickle);
+            python_pickle = upkpickle(&npickle);
             if (subworld) {
                 int size = npickle;
                 nrnmpi_int_broadcast(&size, 1, 0);
-                nrnmpi_char_broadcast(s, size, 0);
+                nrnmpi_char_broadcast(python_pickle, size, 0);
             }
         } else {
-            s = upkstr();
+            char* fname_str = upkstr();
             if (subworld) {
-                int size = strlen(s) + 1;
-                // printf("%d exec hoc fun size = %d\n", nrnmpi_myid_world, size);
+                int size = strlen(fname_str) + 1;
                 nrnmpi_int_broadcast(&size, 1, 0);
-                nrnmpi_char_broadcast(s, size, 0);
+                nrnmpi_char_broadcast(fname_str, size, 0);
             }
-            fname = hoc_lookup(s);
-        }
-        // printf("execute helper style %d fname=%s obj=%s\n", style, fname->name,
-        // hoc_object_name(ob));
-        if (style != 3 && !fname) {
-            fprintf(stderr, "%s not a function in %s\n", s, hoc_object_name(ob));
-            hoc_execerror("ParallelContext execution error", 0);
+            fname = hoc_lookup(fname_str);
+            if (!fname) {
+                fprintf(stderr, "%s not a function in %s\n", fname_str, hoc_object_name(ob));
+                hoc_execerror("ParallelContext execution error", 0);
+            }
+            delete[] fname_str;
         }
+
         int argtypes = upkint();  // first is least signif
         if (subworld) {
             // printf("%d exec argtypes = %d\n", nrnmpi_myid_world, argtypes);
             nrnmpi_int_broadcast(&argtypes, 1, 0);
         }
-        for (j = argtypes; (i = j % 5) != 0; j /= 5) {
+        for (int i = 0, j = argtypes; (i = j % 5) != 0; j /= 5) {
```

Comment:
Nitpick: any way to get rid of the hardcoded 5 here? I dunno if it's related, but since we added a `std::variant`, can we use `std::variant_size_v` to get the size (probably with a `constexpr`)?

Reply:
I think I will do in another PR otherwise it will make this one too complex

And the `std::variant` is not for those arguments, this is only in `bbslsrv`, right now

Reply:
Okay, noted!